File: geral/livro_entendendo_algoritmos/recursion.rb
def main()
    puts factorial(5)

    puts fibonacci(5)
end

def factorial(n)
    if n >= 2
        return n * factorial(n-1)
    end
    
    return 1
end

def fibonacci(n)
    if n >= 3
        return fibonacci(n - 1) + fibonacci(n - 2)
    end

    return 1
end

main

File: geral/livro_entendendo_algoritmos/recursion/recursion.go
package main

import "fmt"

func main() {
	fmt.Println(factorial(6))

	fmt.Println(fibonacci(6))
}

func factorial(n int) int {
	if n >= 2 {
		return n * factorial(n-1)
	}

	return 1
}

func fibonacci(n int) int {
	if n >= 3 {
		return fibonacci(n-1) + fibonacci(n-2)
	}

	return 1
}
